File: LeetCode/0069.go
func mySqrt(x int) int {
	if x < 0 {
		return -1
	}
	var l, r, tmp, mid int64 = 0, int64(x) + 1, 0, 0
	for l < r {
		mid = l + (r-l)/2
		tmp = mid * mid
		if tmp <= int64(x) {
			l = mid + 1
		} else {
			r = mid
		}
	}
	return int(l - 1)
}

File: leetcode/0069.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x < 0) return -1;
        long l = 0, r = (long)x + 1;
        while (l < r) {
            int mid = l + (r - l) / 2;
            long temp = (long)mid * (long)mid;
            if (temp <= x) {
                l = mid + 1;
            } else {
                r = mid;
            }
        }
        return l - 1;
    }
};